contracts/test/TestJBveBanny.t.sol

```diff
@@ -60,55 +68,131 @@ contract JBveBannyTests is TestBaseWorkflow {
+    assert(_useJbToken);
+    assert(!_allowPublicExtension);
     assertEq(_jbveBanny.ownerOf(1), _projectOwner);
-    (uint256 amount, uint256 duration, , bool isJbToken,) = _jbveBanny.getSpecs(1);
+    (uint256 amount, uint256 duration, , bool isJbToken, ) = _jbveBanny.getSpecs(1);
     assertEq(amount, 10 ether);
-    assertEq(duration, 864000);
+    assertEq(duration, 604800);
     assert(isJbToken);
   }
 
   function testUnlockingTokens() public {
-    IJBToken _token = mintIJBTokens();
-    _jbveBanny.lock(_projectOwner, 10 ether, 864000, _projectOwner, true, false);
-    (, , uint256 lockedUntil, ,) = _jbveBanny.getSpecs(1);
-    evm.warp(lockedUntil * 2);
+    IJBToken _jbToken = mintIJBTokens();
+    _jbveBanny.lock(_projectOwner, 10 ether, 604800, _projectOwner, true, false);
+    (, , uint256 lockedUntil, , ) = _jbveBanny.getSpecs(1);
+    evm.warp(lockedUntil + 2);
     _jbveBanny.approve(address(_jbveBanny), 1);
     JBUnlockData[] memory unlocks = new JBUnlockData[](1);
-    unlocks[0] = JBUnlockData(
-      1, _projectOwner
-    );
+    unlocks[0] = JBUnlockData(1, _projectOwner);
     _jbveBanny.unlock(unlocks);
-    assertEq(_token.balanceOf(address(_jbveBanny), _projectId), 0);
+    assert(_jbveBanny.tokenVotingPowerAt(1, block.number) == 0);
+    (int128 _amount, uint256 end, , , ) = _jbveBanny.locked(1);
+    assertEq(_amount, 0);
+    assertEq(end, 0);
+    assertEq(_jbToken.balanceOf(address(_jbveBanny), _projectId), 0);
   }
 
   function testExtendLock() public {
     mintIJBTokens();
-    uint256 _tokenId = _jbveBanny.lock(_projectOwner, 10 ether, 864000, _projectOwner, true, false);
-    (, uint256 d, uint256 lockedUntil, ,) = _jbveBanny.getSpecs(_tokenId);
-    assertEq(d, 864000);
-    evm.warp(lockedUntil * 2);
+    uint256 _tokenId = _jbveBanny.lock(_projectOwner, 10 ether, 604800, _projectOwner, true, false);
+    (, uint256 d, uint256 lockedUntil, , ) = _jbveBanny.getSpecs(_tokenId);
+    assertEq(d, 604800);
+    evm.warp(lockedUntil + 2);
+    uint256 votingPowerBeforeExtending = _jbveBanny.tokenVotingPowerAt(1, block.number);
 
     JBLockExtensionData[] memory extends = new JBLockExtensionData[](1);
-    extends[0] = JBLockExtensionData(
-      1, 8640000
-    );
+    extends[0] = JBLockExtensionData(1, 2419200);
     _tokenId = _jbveBanny.extendLock(extends)[0];
-
-    (, uint256 duration, , ,) = _jbveBanny.getSpecs(_tokenId);
-    assertEq(duration, 8640000);
+    uint256 votingPowerAfterExtending = _jbveBanny.tokenVotingPowerAt(1, block.number);
+    assert(votingPowerAfterExtending > votingPowerBeforeExtending);
+    (, uint256 duration, , , ) = _jbveBanny.getSpecs(_tokenId);
+    assertEq(duration, 2419200);
   }
 
+
+  function testRedeem() public {
+    mintIJBTokens();
+    uint256 _tokenId = _jbveBanny.lock(_projectOwner, 10 ether, 604800, _projectOwner, true, false);
+    (, , uint256 lockedUntil, , ) = _jbveBanny.getSpecs(_tokenId);
+    evm.warp(lockedUntil * 2);
+    evm.stopPrank();
```

Comment:
move the startPrank in this function (or the stop in mint), would increase clarity+++